aptos-node/src/storage.rs

```diff
@@ -42,10 +46,15 @@ pub(crate) fn bootstrap_db(
             },
             Either::Right(fast_sync_db_wrapper) => {
                 let temp_db = fast_sync_db_wrapper.get_temporary_db_with_genesis();
-                maybe_apply_genesis(&DbReaderWriter::from_arc(temp_db), node_config)?;
-
+                let ledger_info_opt =
+                    maybe_apply_genesis(&DbReaderWriter::from_arc(temp_db), node_config)?;
                 let (db_arc, db_rw) = DbReaderWriter::wrap(fast_sync_db_wrapper);
                 let fast_sync_db = db_arc.get_fast_sync_db();
+                // FastSyncDB requires ledger info at epoch 0 to establish provenance to genesis
+                if let Some(ledger_info) = ledger_info_opt {
+                    fast_sync_db.commit_genesis_ledger_info(&ledger_info)?;
+                }
```

Comment:
At this point the main DB must have the genesis LI. Shall we just read it out and put it to the fast sync DB unconditionally / if it doesn't present already?

Reply:
if the temp_db doesn't apply the genesis, not sure this is a clean start so that we don't apply genesis to the fast_synced_db

Reply:
It's fine, right? The genesis LI is deterministic. If you are paranoid, write only if it doesn't exist in the fast sync DB.

Reply:
addressed